akka-discovery/src/main/scala/akka/discovery/ServiceDiscovery.scala

```diff
@@ -43,8 +67,23 @@ object ServiceDiscovery {
       (t.address, t.host, t.port)
     }
 
-    def apply(host: String, port: Option[Int]): ResolvedTarget =
-      ResolvedTarget(host, port, Try(InetAddress.getByName(host)).toOption)
+    /**
+     * @param host the hostname or the IP address of the target
+     * @param port optional port number
+     * @param address IP address of the target. This is used during cluster bootstap when available.
+     */
+    def apply(host: String, port: Option[Int]): ResolvedTarget = {
+      // FIXME API should we really have this? Isn't InetAddress.getByName doing blocking DNS lookup?
```

Comment:
^^^ ?

Reply:
I agree that this is unnecessary and undesirable. It seems to only be used as a convenience from one test. I guess the intention is that this is only used for addresses that have been resolved already, but if it's a public API then that's not safe to assume.

If a two-argument convenience is wanted, it's better to take `address` and `port` and derive `host` from `address` rather than the other way around.

Reply:
I was wrong about it only being used in one test. It's also used in several places in Akka Management.

Reply:
ok, I'll look into those places and replace them there before releasing 0.50 (so that it will not break when updating to Akka 2.5.20)

Reply:
fixed: https://github.com/akka/akka-management/pull/428/commits/81d1165fad25b9ce39099ff66c5e796aa9ec19d1